straxen/plugins/acqmon_processing.py

```diff
@@ -240,6 +242,160 @@ def compute(self, aqmon_hits, start, end):
         result = result[sort]
         return result
 
+@export
+class GPS_sync(strax.Plugin):
+    """
+    Correct the event times to GPS time. 
+      1. Finds the TTL GPS pulses coming into the AM from the gps 
+    module and their pairs in the ASCII file coming from the module
+    for the correspondant run.
+      2. Corrects the timestamp of all events by linearly interpolating
+    between the previous and next sync pulses. 
+    """
+        
+    __version__ ='0.1.0'
+    depends_on = ('aqmon_hits', 'event_info')
+    provides  = 'gps_sync'
+    data_kind = 'gps_sync'
+    
+    def infer_dtype(self):
+        dtype = [(('Time since unix epoch [ns]', 'time'), np.int64),
+                 (('GPS absolute time [ns]', 't_gps'), np.int64),
+                ]
+        return dtype
+
+
+    def gps_times_from_runid(self,run_id):
+        '''
+        Fetches the mongodb looking for the gps_sync collection for 
+        timestamps between the start and end times of a given run_id.
+        '''
+
+        rundb = utilix.xent_collection()
+        gps_times = utilix.xent_collection(collection='gps_sync',  database='xenonnt')
+
+        if isinstance(run_id, str):
+            run_id = int(run_id)
+
+        query = {"number": run_id}
+
+        doc = rundb.find_one(query, projection={'start': 1, 'end': 1})
+        
+        start_t = doc['start'].replace(tzinfo=datetime.timezone.utc).timestamp()
+        end_t = doc['end'].replace(tzinfo=datetime.timezone.utc).timestamp()
+        
+        query = {"gps_sec": { '$gte': start_t - 1, '$lte': end_t + 1}, 'channel': 0 }
```

Comment:
If, rather than `start-1` and `stop+1` you add/subtract 10 (or even 11), you should be guaranteed* to get the first and last GPS pulses outside the run. I think this might help with L334 and L342.

* At least since the new GPS reader script, this might not be true for the SR0 data.

Reply:
This was actually implemented in an earlier version when reading from files but from the collection I asusmed would not be needed. I'm not against bring it back, it might help recover some runs but will force to check if any pulses need to be cut, using `cut_outside_run` from L291.